cpp/daal/src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -154,10 +170,14 @@ services::Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::compute(c
 
     for (uint32_t qblock = 0; qblock < nQueryBlockCount; qblock++)
     {
+
         Range curQueryRange = Range::createFromBlock(qblock, maxQueryBlockRowCount, nQueryRows);
         BlockDescriptor<algorithmFpType> queryRows;
         DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(curQueryRange.startIndex, curQueryRange.count, readOnly, queryRows));
         auto curQuery = queryRows.getBuffer();
+        // Collect sums of squares from train data
+        auto querySumResult = math::SumReducer::sum(math::Layout::RowMajor, curQuery, curQueryRange.count, nFeatures, st);
+        DAAL_CHECK_STATUS_VAR(st);
```

Comment:
May reduce performance for classification

Reply:
Sure but it can be fixed only introducing another function